### DISCUSSION TITLE
when building a bundle, can i sign bundle remotely with a inner company sign server?

### DISCUSSION THREAD

**Discussion by andyliu-666** [rauc/rauc#1181] [Q&A]
when building a bundle, can i sign bundle remotely with a inner company sign server?
---i dont want to sign bundle locally by give --cert / --key in cmd "rauc bundle".

**Comment by jluebbe** [comment]
I'm not sure I understand your question.

For signing with keys which are not available as files, you can use PKCS#11 URIs instead of `--key` to use a key on a (local or network) HSM.

Furthermore, there is support for external signing. To use this, you need to sign with a temporary key first during `rauc bundle`. This signature can then be replaced later: https://rauc.readthedocs.io/en/latest/reference.html#external-signing-and-pki

**Reply by andyliu-666** [reply]
thanks for your answer.
so I can't replace the sign process in bundle creation with my remote sign server. 
the only way is re-sign with my remote sign server after the bundle is created.
